tests/core/test_repo.py

```diff
@@ -1,179 +1,190 @@
-#
-#     current_version = "zenml_" + str(__version__)
-#
-#     bogus_step = "asdfghjklöä"
-#
-#     step_versions = repo.get_step_versions_by_type(random_step)
-#
-#     assert step_versions == {current_version}
-#
-#     assert repo.get_step_versions_by_type(bogus_step) is None
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+# TODO: [MEDIUM] add basic CRUD tests back in
+
+import os
+from pathlib import Path
+
+import pytest
+from git.repo.base import Repo
+
+import zenml
+from zenml.core.base_component import BaseComponent
+from zenml.core.constants import ZENML_DIR_NAME
+from zenml.core.git_wrapper import GitWrapper
+from zenml.core.local_service import LocalService
+from zenml.core.repo import Repository
+from zenml.stacks.base_stack import BaseStack
+
+# a way to get to the root
+ZENML_ROOT = str(Path(zenml.__path__[0]).parent)
+TEST_ROOT = os.path.join(ZENML_ROOT, "tests")
+
+
+def test_repo_init_from_empty_directory_raises_error(tmp_path: str) -> None:
+    """Check empty directory"""
+    with pytest.raises(FileNotFoundError):
+        _ = Repository(os.path.join(tmp_path, "empty_repo"))
+
+
+def test_initializing_repository_from_root_sets_cwd_as_repo_path() -> None:
+    """Check initializing repository from root sets current directory
+    as the ZenML repository path"""
+    current_dir = os.getcwd()
+    os.chdir("/")
+    repo = Repository()
+    assert repo.path == "/"
+    os.chdir(current_dir)
+
+
+def test_initializing_repository_without_git_repo_does_not_raise_error(
+    tmp_path: str,
+) -> None:
+    """Check initializing repository without git repository does not raise error"""
+    repo = Repository(str(tmp_path))
+    assert repo.git_wrapper is None
+
+
+def test_initializing_repo_with_git_repo_present_sets_git_wrapper(
+    tmp_path: str,
+) -> None:
+    """Check initializing repository with git repository sets git wrapper"""
+    git_repo_instance = Repo.init(tmp_path)
+    repo = Repository(str(tmp_path))
+    assert repo.git_wrapper is not None
+    assert repo.git_wrapper.repo_path == str(tmp_path)
+    assert repo.git_wrapper.git_repo == git_repo_instance
+
+
+def test_initializing_repo_sets_up_service(tmp_path: str) -> None:
+    """Check initializing repository sets up service"""
+    repo = Repository(str(tmp_path))
+    assert repo.service is not None
+    assert isinstance(repo.service, BaseComponent)
+    assert isinstance(repo.service, LocalService)
+
+
+def test_repo_double_init(tmp_path: str) -> None:
+    """explicitly constructing another repository should fail"""
+    _ = Repo.init(tmp_path)
+    os.mkdir(os.path.join(tmp_path, ZENML_DIR_NAME))
+
+    with pytest.raises(Exception):
```

Comment:
Same as above, I think we should specify the Exceptions here and in the tests below